scripts/extensions/pool_update.apply

```diff
@@ -36,16 +36,25 @@ def failure_message(code, params):
     return xmlrpclib.dumps((rpcparams, ), '', True)
 
 
-def execute_apply(session, update_package, yum_conf_file, nogpgcheck):
-    FNULL = open(os.devnull, 'w')
-    retcode = subprocess.call('yum -c ' + yum_conf_file + ' install -y ' +
-                              nogpgcheck + update_package, shell=True, stdout=FNULL, stderr=FNULL)
-    if retcode != 0:
-        raise ApplyFailure(
-            'Failed to install update_package (%r).' % update_package)
+def execute_apply(session, update_package, yum_conf_file):
+    cmd = ['yum', 'install', '-y', '--noplugins', '-c', yum_conf_file, update_package]
+    p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, close_fds=True)
+    p.wait()
+    output = p.stdout.read()
+    xcp.logger.info('pool_update.apply %r returncode=%r output=%r', cmd, p.returncode, output)
```

Comment:
If we use `p.communicate()`, why don't we set `stdout=None` (default) and `stderr=supprocess.PIPE` and `_, output = p.communicate()`? It should also make sure that the output is small, if any.